rbd/watchers_mimic.go

```diff
@@ -0,0 +1,98 @@
+// +build !luminous,!mimic
+//
+// Ceph Nautilus is the first version that supports watchers through librbd.
+
+package rbd
+
+// #cgo LDFLAGS: -lrbd
+// #include <errno.h>
+// #include <rbd/librbd.h>
+import "C"
+
+// Handle for a watcher
+// This type was added so that Image.HasWatchers() can be tested.
+type rbdUpdateHandle struct {
+	handle C.uint64_t
+}
+
+// TODO: rename to UpdateWatch() and accept callback arguments
+// This function was added so that Image.HasWatchers() can be tested.
+//
+// Implements:
+//   int rbd_update_watch(rbd_image_t image, uint64_t *handle,
+//                        rbd_update_callback_t watch_cb, void *arg)
+func (image *Image) updateWatch() (*rbdUpdateHandle, error) {
+	if err := image.validate(imageIsOpen); err != nil {
+		return nil, err
+	}
+
+	ruh := &rbdUpdateHandle{}
+
+	// TODO: passing callbacks from Go to C is tricky, see
```

Comment:
It would be nice to make use of the callback infrastructure that #177 provides. :-)
I'm happy to put whatever stuff we need in for private interfaces but it would be even better to have a more complete implementation for consumers. Let me know if you want to do a chat / call to dive deeper into that.

Reply:
Yes, I want to spend some more time on reviewing #177 and adapting the pattern here. So for now it is all private, until a satisfying callback 'standard' is available :+1: